tools/setup-vendor-flutter

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+this_dir=${BASH_SOURCE[0]%/*}
+
+# shellcheck source=tools/lib/git.sh
+. "${this_dir}"/lib/git.sh
+
+
+# One-time setup to let git submodule + flutter SDK know that we're
+# using the main channel (main branch).
+# We should do a quick check that there's no changes.
+if ! submodule_is_clean; then
+    echo >&2 "There are changes in the submodule or it is not initialized."
+    echo >&2 "Please run 'git submodule update --init' then run this script again."
+    exit 1
+fi
+
+git -C vendor/flutter checkout -B main HEAD
```

Comment:
Have you found any behavior that's affected if we leave this out, other than `flutter upgrade`?

If not, it'd be good to leave it out of the initial setup instructions — one less step to take, and in particular brings it back down to zero steps that are entirely our own custom scripts instead of things that are widely used and may be familiar. Then it can just be part of the instructions around `flutter upgrade` instead, which is something a much smaller number of our contributors need to think about.

Reply:
Yes, I found that `flutter doctor` complains that the flutter SDK isn't set up correctly. And since `flutter doctor` would be used by newcomers a lot, it would be confusing to those users.
For reference, the output from `flutter doctor`:
```
[!] Flutter (Channel [user-branch], 3.21.0-12.0.pre.25, on macOS 14.4 23E214 darwin-x64, locale en-SG)
    ! Flutter version 3.21.0-12.0.pre.25 on channel [user-branch] at /Users/chris/code/zulip-flutter/vendor/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at
      https://flutter.dev/docs/get-started/install.
    ! Upstream repository unknown source is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss this error.
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.2)
[✓] IntelliJ IDEA Community Edition (version EAP IC-241.14494.17)
[✓] IntelliJ IDEA Ultimate Edition (version EAP IU-241.14494.17)
[✓] VS Code (version 1.87.2)
[✓] VS Code (version 1.81.0-insider)
[✓] Connected device (4 available)
[✓] Network resources
! Doctor found issues in 1 category.
```

Reply:
Cool, that reasoning makes sense to me.